src/components/editMeasure/testCases/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/resource/ResourceEditor.test.tsx

```diff
@@ -170,4 +170,75 @@ describe("ResourceEditor", () => {
     userEvent.click(actionCenter);
     expect(screen.getByTestId("elements-copy")).not.toBeInTheDocument;
   });
+
+  it("opens AddElementDialog, interacts with it, and can close it", async () => {
+    const setInitialFormikValuesStu6 = jest.fn();
+    const setValidationSchema = jest.fn();
+    const mockDispatch = jest.fn();
+
+    render(
+      <QiCoreResourceContext.Provider
+        value={{ state: mockPatientState, dispatch: mockDispatch }}
+      >
+        <ResourceEditor
+          selectedResourceID="6fb9d817-76c5-4b68-ba06-92c7429e6b5c"
+          setValidationSchema={setValidationSchema}
+          setInitialFormikValuesStu6={setInitialFormikValuesStu6}
+          selectedResource={mockSelectedResource}
+          onCancel={mockOnCancel}
+          canEdit={true}
+        />
+      </QiCoreResourceContext.Provider>
+    );
+
+    // Click the "Add Attribute(s)" button to open dialog
+    const addAttributeButton = screen.getByTestId(
+      "add-attribute-dialog-button"
+    );
+    userEvent.click(addAttributeButton);
+
+    // Verify dialog is open
+    await waitFor(() => {
```

Comment:
I don't think we need waitFor, because you haven't used them in consequent dialog openings and they seem to work fine. Also it is ideal to use "findBy" instead of "getBy" before we can introduce waitFor